packages/eslint-plugin-pf-codemods/src/rules/v6/emptyStateHeaderMoveIntoEmptyState/emptyStateHeader-move-into-emptyState.ts

```diff
@@ -0,0 +1,261 @@
+        case "JSXExpressionContainer":
+        case "JSXSpreadChild":
+          return getNodesText(
+            children.map(
+              (child) =>
+                (child as JSXExpressionContainer | JSXSpreadChild).expression
+            )
+          );
+        case "JSXElement":
+        case "JSXFragment":
+          return getNodesText(children as JSXElement[] | JSXFragment[]);
+        default:
+          return "";
+      }
+    };
+
+    return {
+      ImportDeclaration(node: ImportDeclaration) {
+        if (
+          !pfPackageMatches(pkg, node.source.value) ||
+          !includesImport(node.specifiers, "EmptyStateHeader")
+        ) {
+          return;
+        }
+      },
+      JSXElement(node: JSXElement) {
+        if (!isComponentNode(node, "EmptyState")) {
+          return;
+        }
+
+        const header = getChildElementByName("EmptyStateHeader", node);
+
+        if (!header || header.type !== "JSXElement") {
+          return;
+        }
+
+        const headingClassNameAttribute = getAttribute(header, "className");
+        const headingLevelAttribute = getAttribute(header, "headingLevel");
+        const titleClassNameAttribute = getAttribute(header, "titleClassName");
+        const titleTextAttribute = getAttribute(header, "titleText");
+        const headerIconAttribute = getAttribute(header, "icon");
+
+        const headerChildren = header.children;
+
+        if (!titleTextAttribute && !headerChildren.length) {
+          context.report({
+            node,
+            message: `${baseMessage} You must manually supply a titleText prop, then you can rerun this codemod.`,
+          });
+          return;
+        }
+
+        if (titleTextAttribute && headerChildren.length) {
+          context.report({
+            node,
+            message: `${baseMessage} Because the children for EmptyStateHeader are now inaccessible you must remove either the children or the titleText prop, then you can rerun this codemod.`,
+          });
+          return;
+        }
+
+        const headingClassNameValue = getAttributeValueText(
+          headingClassNameAttribute
+        );
+
+        const headingClassName = headingClassNameValue
+          ? `headerClassName="${headingClassNameValue}"`
+          : "";
+        const headingLevel = getAttributeText(headingLevelAttribute);
+        const titleClassName = getAttributeText(titleClassNameAttribute);
+        const titleTextPropValue = getAttributeText(titleTextAttribute);
+
+        const titleText =
+          titleTextPropValue ||
+          `titleText="${getElementChildText(headerChildren)}"`;
+
+        const iconPropValue = getExpressionValue(headerIconAttribute?.value);
+
+        const emptyStateIconComponent =
+          iconPropValue?.type === "JSXElement" ? iconPropValue : undefined;
+
+        const emptyStateIconComponentIconAttribute =
+          emptyStateIconComponent &&
+          getAttribute(emptyStateIconComponent, "icon");
```

Comment:
Follow up should include getting an icon that isn't passed to EmptyStateIcon e.g.

```
  <EmptyState>
    <EmptyStateHeader
      titleText='Empty state'
      headingLevel='h4'
      icon={<CubesIcon />}
    />
  </EmptyState>
```

Reply:
Made #595 to address